### PR TITLE
Fix first load of term validation and provide score message and routing.

### PR DESCRIPTION
**Pull request by Saeedla**

### CODE REVIEW

File: src/component/term/TermDetail.tsx [hunk 1]
```diff
@@ -80,21 +80,21 @@ export class TermDetail extends EditableComponent<TermDetailProps, TermDetailSta
...
-        this.props.loadTerm(termName, {fragment: vocabularyName, namespace});
+        this.props.loadTerm(termName, {fragment: vocabularyName, namespace}).then( () => this.loadValidationResults());
     }
 
     private loadValidationResults = () => {
         (this.props.validationResults && this.props.validationResults[this.props.vocabulary.iri]) ?
-        this.computeScore(this.props.validationResults[this.props.vocabulary.iri].filter(result => result.term.iri === this.props.term?.iri)) : this.setBadgeColor(-1);
-    }
+         this.computeScore(this.props.validationResults[this.props.vocabulary.iri].filter(result => result.term.iri === this.props.term?.iri)) : this.setBadgeColor(-1);
+    };
```
Comment (psiotwo) [inline src/component/term/TermDetail.tsx]: Would not 'null' value be meaningful instead of -1 for such cases?

**Review by psiotwo — APPROVED** [review]
lgtm

File: src/component/term/TermDetail.tsx (hunk 1 above)
Comment (Saeedla) [reply src/component/term/TermDetail.tsx]: Changes committed.

**Review by psiotwo — COMMENTED** [review]
code-wise LGTM, but now I cannot see the badge at all in
https://deploy-preview-26--termit-dev.netlify.app/#/vocabularies/sb-z-2006-183/terms/testpk?namespace=http://onto.fel.cvut.cz/ontologies/slovnik/

**Comment by Saeedla** [comment]
> code-wise LGTM, but now I cannot see the badge at all in
> https://deploy-preview-26--termit-dev.netlify.app/#/vocabularies/sb-z-2006-183/terms/testpk?namespace=http://onto.fel.cvut.cz/ontologies/slovnik/

Fixed.

**Review by psiotwo — COMMENTED** [review]
Now things work when vocabulary is loaded first (i.e. I open a vocabulary, then I go to term detail).

But when I open a term directly (e.g. from the last edited assets on the dashboard) from a vocabulary which is not loaded yet, the badge is missing.

image

**Review by psiotwo — APPROVED** [review]
lgtm

**Pull request merged**